test/integration/nodeagent/operatingsystemconfig/operatingsystemconfig_test.go

```diff
@@ -264,14 +292,17 @@ var _ = Describe("OperatingSystemConfig controller tests", func() {
 		// disable unit4 and remove all drop-ins
 		// remove only first drop-in from unit5
 		// file1 and unit3 are unchanged.
+		// move file3 from unit.files to files while keeping it unchanged
```

Comment:
- Let's add a unit file for `unit3` that is not changed to ensure such a unit is not unnecessarily restarted
- Let's add a unit file for another unit that is changed to ensure such a unit is restarted (today, we only check that a unit is restarted when all unit files are removed)